pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/ReportingJobService.kt

```diff
@@ -63,15 +63,16 @@ class ReportingJobService: JobService() {
       return b
     }
 
-    fun fromBundle(bundle: Bundle): ReportEvent {
-      val event = ReportEventType.valueOf(bundle.getString(BUNDLE_EVENT_TYPE_KEY))
-      when (event) {
+    fun fromBundle(bundle: Bundle): ReportEvent? {
```

Comment:
The biggest difference is making this return a `ReportEvent?`. So we no longer have the expectation that the bundle will be correctly formed 100% of the times.